src/hotspot/cpu/aarch64/c1_LIRAssembler_aarch64.cpp

```diff
@@ -2984,7 +2984,25 @@ void LIR_Assembler::membar_loadstore() { __ membar(MacroAssembler::LoadStore); }
 void LIR_Assembler::membar_storeload() { __ membar(MacroAssembler::StoreLoad); }
 
 void LIR_Assembler::on_spin_wait() {
-  Unimplemented();
+  switch (VM_Version::pause_impl_desc().inst()) {
+    case NOP:
```

Comment:
Again, please push this into a macro, called from c1 and c2, that does the right thing for the current machine.